particles/Tutorial/Javascript/2_BasicTemplates/BasicTemplates.arcs

```diff
@@ -1,8 +1,8 @@
 // Tutorial: Basic Templates
 // Shows how to use templates to render data. This recipe has the exact same structure as the Hello World tutorial, only the JS is different.
 
-particle BasicTemplateParticle in 'basic-template.js'
-  consume root
+particle BasicTemplateParticle in 'BasicTemplate.js'
+  root: consumes Slot
```

Comment:
@Cypher1 Sarah tells me that `Slot` here is optional (and probably in the other example above, and some but not all below), and if we don't need it, why not remove it. Is/should it be optional? What is best practice here from a code lab tutorial, we're guiding people, perspective?

Reply:
In the old syntax optional was default, but (if a particle had slots) at least one (consumed) slot would have to resolve.

Making it required here (the default in the new syntax) shouldn't change anything, as there is no alternative slot and the particle has slots.

Additional, optional slots can be marked with `consumes?` or `provides?` to make them optional.

If the slot is not needed, we could remove it but my guess is that it is needed.

Reply:
My take on it is that we should make it non optional.

Firstly, you can already consume/provide a `Slot` vs `[Slot]`, and in the future we will likely make slots more parametrized - the rendering of the Android notification is different type than rendering of HTML. Having a `default` will stop making sense at that point.